models/commentsModel.js

```diff
@@ -0,0 +1,9 @@
+const db = require("../db/connection.js");
+const selectCommentsByArticleId = (article_id) => {
+  const query_str = `SELECT comment_id,votes,created_at,author,body FROM comments WHERE article_id =$1 ORDER BY created_at DESC;`;
+  return db.query(query_str, [article_id]).then(({ rows: comments }) => {
+    return comments;
```

Comment:
You have a test for a 404 endpoint when the article ID is valid but NOT existing and I am unsure as to how this test is passing as we are not handling it inside the model. At the moment if passed an article_id of 99999, we would just return an empty array. We would also return an empty array for an article with no comments.  

We need some kind of way of checking, if we get an empty array, whether the article actually exists. Please ammend this and let me know once you have done so, so that I can have another look

Reply:
A the moment In line 7 of commentsController.js , i am calling   selectArticleById(article_id) and its where 404 end point gets handled as i called function to selectCommentsByArticleId() inside the then block after selectArticleById().So comments get only selected if article id is found in db.